R/update_package_object.R

```diff
@@ -0,0 +1,59 @@
+#' Update a data object and accompanying package resource map
+#'
+#' This function updates an data object, and then automatically
+#' updates the package resource map with the new data pid. It
+#' is a convenience wrapper around \code{\link{arcticdatautils::update_object}}
+#' and \code{\link{arcticdatautils::update_resource_map}}.
+#'
+#' @param mn (MNode) Member node
+#' @param data_pid (character) PID for data object to update
+#' @param new_data_path (character) Path to new data object
+#' @param resource_map_pid (character) PID for resource map to update
+#' @param format_id (character) Optional. The format ID to set for the object.
+#' When not set, \code{\link{arcticdatautils::guess_format_id}} will be used
+#' to guess the format ID. Should be a \href{https://cn.dataone.org/cn/v2/formats}{DataONE format ID}.
+#' @param public (logical) Optional. Make the update public. If FALSE,
+#' will set the metadata and resource map to private (but not the data objects).
+#' This applies to the new metadata PID and its resource map and data object.
+#' Access policies are not affected.
+#'
+#' @keywords update_object update_resource_map
+#'
+#' @export
+
+update_package_object <- function(mn,
+                           data_pid,
+                           new_data_path,
+                           resource_map_pid,
+                           format_id = NULL,
+                           public = FALSE) {
+
+    #argument checks:
+    library(assertthat)
+    library(arcticdatautils)
+    assertthat::assert_that(is(mnTest, "MNode"),
```

Comment:
`assertthat` is a good package but you might consider not making use of it unless it offers a significant benefit to your package's users over `stopifnot` or other base R ways to verify arguments. Making use of an extra package in your package will require your users to install that extra package when they install your package. This isn't a problem when you add one or two extra packages but it can make your package slow to install if you add a lot.

Whichever extra packages you keep need to get added to your DESCRIPTION file in either the Imports or Suggests fields. See http://r-pkgs.had.co.nz/namespace.html